### ISSUE TITLE
Crashes and bugs behind a proxy

### ISSUE THREAD

**Issue by Bob0111** [PaulCombal/SamRewritten-legacy#114]
When I finish the compile, and try to start the launcher with the scripts, I just get an error message:
arsing error: parse error: premature EOF
                                       
                     (right here) ------^
I have already installed the required package and no problems about the compiling.

**Comment by PaulCombal** [comment]
Hi, are you connected to the internet? Are you behind a proxy?

**Comment by Bob0111** [comment]
> Hi, are you connected to the internet? Are you behind a proxy?

Yes. I'm behind a proxy and unable to disable that or I cannot use the steam. By the way, I set the proxy through the kde plasma setting. I will use the proxychains trying to use the software again.
Edit: try with the proxychains, still the same error.

**Comment by PaulCombal** [comment]
Can you tell me what you have in your `~/.cache/SamRewritten` ?

**Comment by Bob0111** [comment]
> Can you tell me what you have in your `~/.cache/SamRewritten` ?

game_list.json. But it is an empty file.

**Comment by PaulCombal** [comment]
SamRewritten needs to download the list of apps before launching, if you're behind a proxy it won't be able to do so. Can you download `https://raw.githubusercontent.com/PaulCombal/SteamAppsListDumps/master/game_achievements_list.json`, save it to `~/.cache/SamRewritten/game_list.json` and see if that works?

In any case this is a bug that should be addressed

**Comment by Bob0111** [comment]
> SamRewritten needs to download the list of apps before launching, if you're behind a proxy it won't be able to do so. Can you download `https://raw.githubusercontent.com/PaulCombal/SteamAppsListDumps/master/game_achievements_list.json`, save it to `~/.cache/SamRewritten/game_list.json` and see if that works?
> 
> In any case this is a bug that should be addressed

Thank you! Now the app is just working fine. But the achievement image fails to load. I think in the future it should be able to read the environment path about the proxy and use it. Because sometimes we need to use the proxy to connect the Internet.
